run-services.sh

```
@@ -2,57 +2,156 @@
+      if contains_service "$1" "$ALL_SERVICES"; then
+        TARGET_SERVICES+=("$1")
+      else
+        echo "Error: Unknown service '$1'. Valid options: backend, all, infra, or a specific service ($ALL_SERVICES)"
+        exit 1
+      fi
+      shift
+      ;;
+  esac
+done
+# remove duplicates
+TARGET_SERVICES=($(echo "${TARGET_SERVICES[@]}" | tr ' ' '\n' | sort -u | tr '\n' ' '))
```

Shellcheck (0.10.0)

[warning] Prefer mapfile or read -a to split command output (or quote to avoid splitting).

(SC2207)